### PR TITLE
Re-enable tests disabled in #3880 

### PR DESCRIPTION
**Pull request by yutaroyamanaka**
worked on #3881 

Basically, I followed what @annasong20 [suggested](https://github.com/kubernetes-sigs/kustomize/issues/3881#issuecomment-1445006217)

- For `TestFnContainerTransformer` test, I used `gcr.io/kustomize-functions/e2econtainerconfig` as [this Makefile](https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/internal/commands/e2e/e2econtainerconfig/Makefile) builds. 
- For `TestFnContainerGenerator` test, I chose [`enable-gcp-services`](https://catalog.kpt.dev/enable-gcp-services/v0.1/?id=enable-gcp-services) from [the kpt catalog](https://catalog.kpt.dev/).

### CODE REVIEW

**Comment by koba1t** [comment]
/triage accepted
/kind bug
/priority important-soon

/ok-to-test

cc @annasong20

File: api/krusty/fnplugin_test.go [hunk 1]
```diff
@@ -504,313 +504,94 @@ func skipIfNoDocker(t *testing.T) {
 }
 
 func TestFnContainerGenerator(t *testing.T) {
-	t.Skip("wait for #3881")
 	skipIfNoDocker(t)
-
-	// Function plugins should not need the env setup done by MakeEnhancedHarness
 	th := kusttest_test.MakeHarness(t)
-
-	th.WriteK(".", `
```
Comment (KnVerey) [inline api/krusty/fnplugin_test.go:513]: Is there a reason not to use these test helpers? I think the following should work:

```diff
diff --git a/api/krusty/fnplugin_test.go b/api/krusty/fnplugin_test.go
index 8d3e0e196..1953adf72 100644
--- a/api/krusty/fnplugin_test.go
+++ b/api/krusty/fnplugin_test.go
@@ -507,15 +507,14 @@ func TestFnContainerGenerator(t *testing.T) {
 	skipIfNoDocker(t)
 	th := kusttest_test.MakeHarness(t)
 	o := th.MakeOptionsPluginsEnabled()
-	fSys := filesys.MakeFsOnDisk()
-	b := MakeKustomizer(&o)
 	tmpDir, err := filesys.NewTmpConfirmedDir()
 	assert.NoError(t, err)
-	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "kustomization.yaml"), []byte(`
+
+	th.WriteK(tmpDir.String(), `
 generators:
 - project-service-set.yaml
-`)))
-	assert.NoError(t, fSys.WriteFile(filepath.Join(tmpDir.String(), "project-service-set.yaml"), []byte(`
+`)
+	th.WriteF(filepath.Join(tmpDir.String(), "project-service-set.yaml"), `
 apiVersion: blueprints.cloud.google.com/v1alpha1
 kind: ProjectServiceSet
 metadata:
@@ -528,9 +527,8 @@ spec:
   services:
     - compute.googleapis.com
   projectID: foo
-`)))
-	m, err := b.Run(fSys, tmpDir.String())
-	assert.NoError(t, err)
+`)
+	m := th.Run(tmpDir.String(), o)
 	actual, err := m.AsYaml()
 	assert.NoError(t, err)
 	assert.Equal(t, `apiVersion: serviceusage.cnrm.cloud.google.com/v1beta1
```

File: api/krusty/fnplugin_test.go [hunk 2]
```diff
@@ -504,313 +504,94 @@ func skipIfNoDocker(t *testing.T) {
...
   name: demo
   annotations:
     config.kubernetes.io/function: |
       container:
-        image: gcr.io/kustomize-functions/example-cockroachdb:v0.1.0
+        image: gcr.io/kpt-fn/enable-gcp-services:v0.1.0
 spec:
-  replicas: 3
-`)
-	// Create some additional resource just to make sure everything is added
```
Comment (KnVerey) [inline api/krusty/fnplugin_test.go:532]: Please preserve this aspect of the test, i.e. have an existing resource that should continue to appear in the result

File: api/krusty/fnplugin_test.go [hunk 3]
```diff
@@ -504,313 +504,94 @@ func skipIfNoDocker(t *testing.T) {
...
 }
 
 func TestFnContainerTransformer(t *testing.T) {
-	t.Skip("wait for #3881")
 	skipIfNoDocker(t)
-
-	// Function plugins should not need the env setup done by MakeEnhancedHarness
 	th := kusttest_test.MakeHarness(t)
-
-	th.WriteK(".", `
```
Comment (KnVerey) [inline api/krusty/fnplugin_test.go:723]: Same request to continue to use the test helpers

**Review by KnVerey — COMMENTED** [review]
Thanks for the PR! I left a couple requests in comments

File: api/krusty/fnplugin_test.go (hunk 2 above)
Comment (yutaroyamanaka) [reply api/krusty/fnplugin_test.go:532]: That sounds reasonable. 
I added another resource to assure this aspect when applying a generator.
https://github.com/kubernetes-sigs/kustomize/pull/5111/commits/c9f500cc0b4b038541629bab9e578acdc3921c08

File: api/krusty/fnplugin_test.go (hunk 1 above)
Comment (yutaroyamanaka) [reply api/krusty/fnplugin_test.go:513]: It does make sense. I followed your suggestion.

**Comment by yutaroyamanaka** [comment]
Hi @KnVerey 
Thank you for feedback and  I followed your suggestions.

**Comment by KnVerey** [comment]
Thanks!
/lgtm
/approve

**Comment by yutaroyamanaka** [comment]
@KnVerey 
Thank you for your review.
I confirmed e2e test failed [here ](https://github.com/kubernetes-sigs/kustomize/actions/runs/4583621997/jobs/8123770737) because `gcr.io/kustomize-functions/e2econtainerconfig` doesn't exist in the public anymore.
I fixed this by building an image locally like [`cmd/config/internal/commands/e2e/e2e_test.go`](https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/internal/commands/e2e/e2e_test.go#L789) although this way doesn't look very appropriate. https://github.com/kubernetes-sigs/kustomize/pull/5111/commits/2649d39fd300df34addbbecbe956048a9d0f659d

**Comment by KnVerey** [comment]
> I fixed this by building an image locally like [cmd/config/internal/commands/e2e/e2e_test.go](https://github.com/kubernetes-sigs/kustomize/blob/master/cmd/config/internal/commands/e2e/e2e_test.go#L789) although this way doesn't look very appropriate. https://github.com/kubernetes-sigs/kustomize/commit/2649d39fd300df34addbbecbe956048a9d0f659d

Actually, I think this is a smart approach! It gets rid of a network and registry dependency in the test, and it is not important to test docker's `pull` functionality specifically IMO. We should consider adopting this approach by default for container-based tests, assuming it proves reliable.

/lgtm

**Pull request merged**